### PR TITLE
[enhancement](function) change some function nullable mode

### LINKED ISSUE #27435: [Enhancement] Reduct the unnecessary ALWAYS_NULLABLE in nullable mode
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/doris/issues?q=is%3Aissue) and found no similar issues.

### Description

for many doris' functions, they have nullable mode `ALWAYS_NULLABLE` but in fact no need to be. many of them could be changed to `DEPEND_ON_ARGUMENT` or `CUSTOM` just like https://github.com/apache/doris/pull/27343, https://github.com/apache/doris/pull/22321 and https://github.com/apache/doris/pull/26827. 

### Solution

It may relate to 3 position: BE's function's implementation, function prototype of Nereids planner and function prototype of Legacy planner.
If could reduct it, reduct it.

### Are you willing to submit PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by koarz**
## Proposed changes

Issue Number: close #27435 

change some function nullable mode, and change function unhex ,to_base64, from_base64,will return empty string when input empty string

before:
```
before:
mysql> select to_base64('');
+-----------------+
| to_base64('')   |
+-----------------+
| NULL            |
+-----------------+
mysql> select  from_base64('');
+-----------------+
| from_base64('') |
+-----------------+
| NULL            |
+-----------------+
```
now:
```
mysql> select to_base64('');
+---------------+
| to_base64('') |
+---------------+
|               |
+---------------+
mysql> select  from_base64('');
+-----------------+
| from_base64('') |
+-----------------+
|                 |
+-----------------+
```
## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by koarz** [comment]
run buildall

**Comment by koarz** [comment]
run buildall

**Comment by koarz** [comment]
run buildall

**Comment by koarz** [comment]
run buildall

**Comment by koarz** [comment]
run buildall

**Comment by koarz** [comment]
run buildall

File: be/src/agent/be_exec_version_manager.h
Comment (zclllyybb) [inline be/src/agent/be_exec_version_manager.h:68]: please combine f and g to shorter words. just like "shrink some function's nullable mode"

**Comment by koarz** [comment]
run buildall

**Review by zclllyybb — CHANGES_REQUESTED** [review]
For those change of functions' behaviour but not signature, you should apply them for both old and new version. For the functions which didn't change their signature, we don't need a old version then.

**Comment by koarz** [comment]
run buildall

File: regression-test/data/query_p0/sql_functions/string_functions/test_string_function.out
Comment (zclllyybb) [inline regression-test/data/query_p0/sql_functions/string_functions/test_string_function.out:1]: can u confirm this file's content? seems unrecognizable in github. not sure if it's right

**Comment by zclllyybb** [comment]
If you changed some SQL results. please show some examples in PR's "Proposed changes" about their result before/now.

**Comment by koarz** [comment]
> If you changed some SQL results. please show some examples in PR's "Proposed changes" about their result before/now.

Okay, it's added now.

File: regression-test/data/query_p0/sql_functions/string_functions/test_string_function.out
Comment (zclllyybb) [reply regression-test/data/query_p0/sql_functions/string_functions/test_string_function.out:1]: confirmed

**Review by zclllyybb — APPROVED** [review]
LGTM, the issue could be closed after this merged.

**Review by BiteTheDDDDt — APPROVED** [review]

**Pull request merged**
